### PR TITLE
Launch WCS label modal from wc-admin

### PR DESCRIPTION
**Pull request by Ferdev**
Fixes https://github.com/Automattic/woocommerce-shipping-issues/issues/17.

Adds this functionality to the Shipping Label banner:

- Calls the WooCommerce Services API to accept its Terms of Service and get the URL of the WooCommerce Services assets.
- Loads and adds to the page the WooCommerce Services assets.
- Opens the WooCommerce Services shipping label creation modal.

### Instructions to test:

With WooCommerce Services active, just press the "Create shipping label" button in the banner.

### CODE REVIEW

**Comment by harriswong** [comment]
👀

**Comment by harriswong** [comment]
I am experiencing some errors:

Steps I did
1. Deactivate WCS
2. Delete WCS 
3. Click "Create shipping label"
4. "Create shipping label" button turns busy and no popup
image
5. Confirmed plugin is installed and activated

**Comment by harriswong** [comment]
Ah interesting. I am using `ngrok` but the assets requested are in 
`GET http://localhost:8085/woocommerce-services-1.22.5.js net::ERR_CONNECTION_REFUSED`

**Comment by c-shultz** [comment]
I followed the same steps as Harris and seeing different errors each time. I think there may be some timing issues. One time the assets endpoint 404'd (maybe the request went out before WCS was activated). Another time, the redux store was undefined (I think it got requested before the JS totally loaded)

**Comment by c-shultz** [comment]
>define( 'WOOCOMMERCE_CONNECT_DEV_SERVER_URL', 'http://localhost:8085/' );

@harriswong, you probably have this constant defined in your environment, which is messing with things.

File: client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js [hunk 1]
```diff
@@ -120,6 +128,82 @@ export class ShippingBanner extends Component {
...
+		} );
+		document.body.appendChild( shippingLabelContainer );
+
+		Promise.all( [
+			new Promise( ( resolve, reject ) => {
+				const script = document.createElement( 'script' );
+				script.src = js;
+				script.async = true;
+				script.onload = resolve;
+				script.onerror = reject; //TODO: handle errors
```
Comment (c-shultz) [inline client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js]: Let's make sure we mention this line in our error-handling issue, @fer, so we can clean this comment up

File: client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js [hunk 2]
```diff
@@ -120,6 +128,82 @@ export class ShippingBanner extends Component {
...
+			} ),
+			new Promise( ( resolve, reject ) => {
+				const head = document.getElementsByTagName( 'head' )[ 0 ];
+				const link = document.createElement( 'link' );
+				link.rel = 'stylesheet';
+				link.type = 'text/css';
+				link.href = styles;
+				link.media = 'all';
+				link.onload = resolve;
+				link.onerror = reject; //TODO: handle errors
```
Comment (c-shultz) [inline client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js]: Let's make sure we mention this line in our error-handling issue, @fer, so we can clean this comment up

**Comment by Ferdev** [comment]
Created those issues:

- https://github.com/Automattic/woocommerce-shipping-issues/issues/42
- https://github.com/Automattic/woocommerce-shipping-issues/issues/43
- https://github.com/Automattic/woocommerce-shipping-issues/issues/44

**Pull request merged**